Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1

```diff
@@ -309,50 +314,34 @@ function Register-ServiceFabricApplicationTypeAction
 
     $exceptionRetryEvaluator = {
         param($ex)
-        $appType = Get-ServiceFabricApplicationTypeAction -ApplicationTypeName $ApplicationTypeName -ApplicationTypeVersion $ApplicationTypeVersion
-        # If provisioning not started, retry register
-        if (!$appType)
-        {
-            Write-Host (Get-VstsLocString -Key SFSDK_ApplicationTypeProvisioningNotStarted)
-            return $true
-        }
-
-        # if provisioning started, wait for it to complete
-        if (($appType.Status -eq [System.Fabric.Query.ApplicationTypeStatus]::Provisioning) -or ($appType.Status -eq [System.Fabric.Query.ApplicationTypeStatus]::Unprovisioning))
-        {
-            Write-Host (Get-VstsLocString -Key SFSDK_ApplicationTypeProvisioningStarted)
-            $appType = Wait-ServiceFabricApplicationTypeTerminalStatus -ApplicationTypeName $ApplicationTypeName -ApplicationTypeVersion $ApplicationTypeVersion
-        }
-
-        # if app type got unprovisioned, retry register
-        if (!$appType)
-        {
-            Write-Host (Get-VstsLocString -Key SFSDK_ApplicationTypeUnprovisioned)
-            return $true
-        }
-
-        # if app type is provisioned, bail out
-        if ($appType.Status -eq [System.Fabric.Query.ApplicationTypeStatus]::Available)
+        # If app already created, don't retry
+        if ($ex.GetType().FullName -eq "System.Fabric.FabricElementAlreadyExistsException")
         {
             return $false
         }
 
-        # if provisioning failed, throw and don't retry
-        throw (Get-VstsLocString -Key SFSDK_RegisterAppTypeFailedWithStatus -ArgumentList @($appType.Status, $appType.StatusDetails))
+        return $true
     }
 
     try
     {
         Invoke-ActionWithDefaultRetries -Action $registerAction `
             -RetryMessage (Get-VstsLocString -Key SFSDK_RetryingRegisterApplicationType) `
-            -ExceptionRetryEvaluator $exceptionRetryEvaluator
+            -ExceptionRetryEvaluator $exceptionRetryEvaluator `
+            -RetryableExceptions @("System.Fabric.FabricTransientException", "System.Fabric.FabricElementAlreadyExistsException", "System.TimeoutException")
     }
     catch
     {
         # print cluster health status if registering failed
         Trace-ServiceFabricClusterHealth
         throw
     }
+
+    $timeoutAction = {
+        Unregister-ServiceFabricApplicationTypeAction -ApplicationTypeName $ApplicationTypeName -ApplicationTypeVersion $ApplicationTypeVersion -TimeoutSec $TimeoutSec
```

Comment:
Will the unregister work when the registration is in progress?

Reply:
+1
This should be a best-effor. suppress any error

Reply:
Suppressing now.